exercises/src/test/scala/datastructures/ListSpec.scala

```diff
@@ -0,0 +1,417 @@
+import org.specs2.mutable.Specification
+import org.specs2.ScalaCheck
+
+import fpinscala.datastructures._
+
+class ListSpec extends Specification {
+  //========================================================================================
+  //#3.1
+  "pattern matching example x" should {
+    "result in 3" in {
+      List.x mustEqual 3
+    }
+  }
+
+  //========================================================================================
+  //#3.2
+  "tail" should {
+    "return Nil for an empty List" in {
+      List.tail(List()) mustEqual Nil
+    }
+
+    "return Nil for single-element List" in {
```

Comment:
This is actually a perfect candidate for being a property-based test. You could have `prop { x: Int => List.tail(List(x)) mustEqual Nil }`.